pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -230,22 +230,20 @@ func getTPMapMatchingSpreadConstraints(pod *v1.Pod, nodeInfoMap map[string]*sche
 		if !podMatchesNodeSelectorAndAffinityTerms(pod, node) {
 			return
 		}
-		// Ensure current node's labels contains all topologyKeys in 'constraints'.
-		for _, constraint := range constraints {
-			if _, ok := node.Labels[constraint.TopologyKey]; !ok {
-				return
-			}
-		}
 
 		nodeTopologyMaps := newTopologyPairsMaps()
 		// nodeInfo.Pods() can be empty; or all pods don't fit
-		for _, existingPod := range nodeInfo.Pods() {
-			if existingPod.Namespace != pod.Namespace {
-				continue
+		for _, constraint := range constraints {
+			// Ensure current node's labels contains all topologyKeys in 'constraints'.
+			if _, ok := node.Labels[constraint.TopologyKey]; !ok {
+				return
 			}
-			podLabelSet := labels.Set(existingPod.Labels)
-			for _, constraint := range constraints {
-				ok, err := podMatchesSpreadConstraint(podLabelSet, constraint)
+			pairAdded := false
+			for _, existingPod := range nodeInfo.Pods() {
```

Comment:
If current `node.Labels` match on the 1st constraint's `topologyKey` **only**, we're wasting a large number of cycles on this loop, correct?

Reply:
When you say 'current node.Labels', did you mean existingPod.Labels ?

How would existing code be able to short circuit and avoid the waste ?

Thanks

Reply:
No, I mean node.Labels.

Suppose you have 2 constraints, the 2nd constraint's topology isn't present in the node's labels.
- Without this PR, it directly returns without processing any pod.
- With this PR, it has to process nodeInfo.Pods() and only return when it hits the 2nd constraint.

Reply:
I restored the first loop over constraints.